docs/internal/DistributedArchitectureGuide.md

```diff
@@ -144,15 +144,74 @@ Some concepts are applicable to both cluster and project scopes, e.g. [persisten
 
 ### Translog
 
-(Explain checkpointing and generations, when happens on Lucene flush / fsync)
+[Basic write model]:https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-replication.html
 
-(Concurrency control for flushing)
+It is important to understand first the [Basic write model] of documents:
+documents are written to Lucene in-memory buffers, then "refreshed" to searchable segments which may not be persisted on disk, and finally "flushed" to a durable Lucene commit on disk.
+This means that newly ingested data may be lost if there is an outage before the next persisted commit on disk.
```

Comment:
I think this sentence should be more conditional, and also it would be possible to do this without a translog if we just accepted high latency on write responses. Something like this perhaps?

```suggestion
If this were the only way we stored the data, we would have to delay the response to every write request until after the data had been flushed to disk, which could take many seconds or longer. If we didn't, it'd mean that we would lose newly-ingested data if there were an outage between sending the response and flushing the data to disk.
```